qiskit_experiments/analysis/curve_fitting.py

```diff
@@ -19,11 +19,11 @@
 import numpy as np
 import scipy.optimize as opt
 from qiskit_experiments.exceptions import AnalysisError
-from qiskit_experiments.base_analysis import AnalysisResult
 from qiskit_experiments.analysis.data_processing import filter_data
+from qiskit_experiments.database_service import DbAnalysisResultV1
 
 
-class CurveAnalysisResult(AnalysisResult):
+class CurveAnalysisResult(DbAnalysisResultV1):
```

Comment:
I think this subclass and the custom str/repr can probably be dropped for now. It would be good if the base class could handle pretty printing of the values +/- error for certain result keys.